extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -356,36 +356,7 @@ private void handleAvro(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             //ignore, Apicurio Avro is not in the classpath
         }
 
-        // --- Apicurio Registry 2.x ---
-        try {
-            Class.forName("io.apicurio.registry.serde.avro.AvroKafkaDeserializer", false,
-                    Thread.currentThread().getContextClassLoader());
-            reflectiveClass.produce(
-                    new ReflectiveClassBuildItem(true, true, false,
-                            "io.apicurio.registry.serde.avro.AvroKafkaDeserializer",
-                            "io.apicurio.registry.serde.avro.AvroKafkaSerializer"));
-
-            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, false,
-                    "io.apicurio.registry.serde.strategy.SimpleTopicIdStrategy",
-                    "io.apicurio.registry.serde.strategy.TopicIdStrategy",
-                    "io.apicurio.registry.serde.avro.DefaultAvroDatumProvider",
-                    "io.apicurio.registry.serde.avro.ReflectAvroDatumProvider",
-                    "io.apicurio.registry.serde.avro.strategy.RecordIdStrategy",
-                    "io.apicurio.registry.serde.avro.strategy.TopicRecordIdStrategy"));
-
-            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, false,
-                    "io.apicurio.registry.serde.DefaultSchemaResolver",
-                    "io.apicurio.registry.serde.DefaultIdHandler",
-                    "io.apicurio.registry.serde.Legacy4ByteIdHandler",
-                    "io.apicurio.registry.serde.fallback.DefaultFallbackArtifactProvider",
-                    "io.apicurio.registry.serde.headers.DefaultHeadersHandler"));
-
-            // Apicurio Registry 2.x uses the JDK 11 HTTP client, which unconditionally requires SSL
-            sslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem(Feature.KAFKA_CLIENT));
-
-        } catch (ClassNotFoundException e) {
-            //ignore, Apicurio Avro is not in the classpath
-        }
+        // note: Apicurio Registry 2.x support moved to quarkus-apicurio-registry-avro
```

Comment:
Should we check if Apicurio classes are used, but the extension is not present (Feature not available)?

Reply:
Hmm good point. Is there something we can do, other than fail the build?

Reply:
I would fail the build with a message indicating that it needs a new extension.

Reply:
It seems a little harsh, but ... yea, OK.

Reply:
Done.